theseus/utils/examples/bundle_adjustment/reprojection_error.py

```diff
@@ -0,0 +1,118 @@
+from typing import List, Optional, Tuple
+
+import torch
+
+import theseus as th
+
+from .util import soft_loss_huber_like
+
+
+class ReprojectionError(th.CostFunction):
+    def __init__(
+        self,
+        camera_rotation: th.SO3,
+        camera_translation: th.Point3,
+        focal_length: th.Vector,
+        loss_radius: th.Vector,
+        world_point: th.Point3,
+        image_feature_point: th.Point2,
+        weight: Optional[th.CostWeight] = None,
+        name: Optional[str] = None,
+    ):
+        if weight is None:
+            weight = th.ScaleCostWeight(
+                torch.tensor(1.0).to(dtype=camera_rotation.dtype)
+            )
+        super().__init__(
+            cost_weight=weight,
+            name=name,
+        )
+        self.camera_rotation = camera_rotation
+        self.camera_translation = camera_translation
+        self.loss_radius = loss_radius
+        self.focal_length = focal_length
+        self.world_point = world_point
+        self.image_feature_point = image_feature_point
+
+        self.register_optim_vars(["camera_rotation", "camera_translation"])
+        self.register_aux_vars(
+            ["loss_radius", "focal_length", "image_feature_point", "world_point"]
+        )
+
+    def error(self) -> torch.Tensor:
+        point__cam = (
```

Comment:
Is the double underscore intentional? (I am not familiar with this convention)

Reply:
I took this from Paloma's tactile pushing [code](https://github.com/facebookresearch/theseus/blob/main/theseus/embodied/motionmodel/quasi_static_pushing_planar.py#L62), where it is used to indicate the frame of reference for a variable. I don't know if it's standard or something she came up with, but I like it, seems concise and easy to parse.